src/pages/ProductDetailsPage.tsx

```diff
@@ -0,0 +1,84 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { Breadcrumbs } from '../components/Breadcrumbs';
+import { Loader } from '../components/Loader';
+import { BackButton } from '../components/BackButton';
+import { DetailsProductGallery } from '../components/DetailsProductGallery';
+import { DetailsProductSelect } from '../components/DetailsProductSelect';
+import { Error } from '../components/Error';
+
+import { Details } from '../types/Details';
+import { DetailsProductAbout } from '../components/DetailsProductAbout';
+import { DetailsProductSpecs } from '../components/DetailsProductSpecs';
+import { ProductsSlider } from '../components/ProductsSlider';
+import { randomSequence } from '../utils/detailsUtils';
+import { useFetch } from '../hooks/useFetch';
+
+export const ProductDetailsPage: React.FC = () => {
+  const [details, setDetails] = useState<Details | null>(null);
+  const [isErrorDetails, setIsErrorDetails] = useState(false);
+  const location = useLocation();
+  const { phones, isErrorPhones } = useFetch();
+
+  const pathArray = location.pathname.slice(1).split('/');
+  const productId = pathArray[1];
+  const currentPhone = phones.find(phone => phone.id === productId);
+  const productDetailsPath = currentPhone?.itemId;
+  const productName = currentPhone?.name;
+  const publicPath = process.env.PUBLIC_URL;
+  const infoPath = `${publicPath}/_new/products/${productDetailsPath}.json`;
+  const randomPhones = useMemo(() => randomSequence(phones), [phones]);
```

Comment:
Why you use useMemo for random value?
Remove it
```suggestion
  const randomPhones = randomSequence(phones);
```

Reply:
I think it needs to avoid rerendering on global state change (adding to cart for example). But @serhii-yunnikov may clarify it, if it needs